### ISSUE TITLE
Sound issues with 2.7.4+ release

### ISSUE THREAD

**Issue by vpmedia** [photonstorm/phaser-ce#78]
This Issue is about a bug in the API (Sound).
I think the last modification to the Sound class (https://github.com/photonstorm/phaser-ce/commit/5735dbf1cf320ca4bc587b332c03b86ed97e56aa) resulted in some kind of regression.
I'm trying to play the same audio file sequentially using the forceRestart parameter but I'm noticing silent gaps in the sequence, so the sound is not always playing. 
No problems using 2.7.3.
I will try to create some kind of minimal example.

**Comment by vpmedia** [comment]
I just took a look at the source file, maybe an idea:
line 894 contains the following code before calling `this._sound.stop(0);`
`this._markedToDelete = true;`
but line 556 `this._sound.stop(0);` is not preceded by `this._markedToDelete = true;`
@trpzn Can you tell your opinion please?

**Comment by vpmedia** [comment]
I had to completely revert the commit https://github.com/photonstorm/phaser-ce/commit/5735dbf1cf320ca4bc587b332c03b86ed97e56aa to solve the issue in my game.
If I have a sound file with 1000msec length and I'm firing sound.play calls with forceRestart every 500msec than the sequence breaks because I think the onended signal gets fired too late.
Expected behavior:
Play - End/Cleanup - Play - End/Cleanup
Whats happening now:
Play - Play - End/Cleanup - End/Cleanup

**Comment by vpmedia** [comment]
I had success to scope down the issue.
If I call SoundManager.stopAll(); before a Sound.play() it breaks playing the audio file.

Minimal example (try to click with mouse button repeatedly and notice how 2.7.5 skips sometimes):

```javascript
(function (self) {
    'use strict';
    var game = new Phaser.Game(800, 600, Phaser.CANVAS, null, {
    preload: function preload() {
        game.load.audio('sound', ['assets/test.mp3', 'assets/test.ogg']);
    },
    create: function create() {
        game.scale.scaleMode = Phaser.ScaleManager.SHOW_ALL;
        game.scale.fullScreenScaleMode = Phaser.ScaleManager.SHOW_ALL;
        game.scale.pageAlignHorizontally = true;
        game.scale.pageAlignVertically = true;
        var sound = game.add.audio('sound');
        var isDecoded = false;
        console.log('Decoding ...');
        game.sound.setDecodedCallback([ sound ], function() {
            console.log('Decoded!');
            isDecoded = true;
        }, this);
        game.input.onDown.add(function() {
            if (isDecoded) {
                console.log('Playing ...');
                game.sound.stopAll();
                sound.play('', 0, 1, false, true);
                // sound.restart();
                console.log(sound);
            }            
        }, this);            

    }
});
})(typeof self !== 'undefined' ? self : this);
```

**Comment by vpmedia** [comment]
@trpzn Take a look at if possible, please. Thanks!

**Comment by vpmedia** [comment]
I've uploaded the whole example here:
https://github.com/vpmedia/phaser-ce-examples/tree/master/issue78

**Comment by trpzn** [comment]
I'm seeking for a fix now, thanks for notice the bug and do the test cases

**Comment by samme** [comment]
@vpmedia Could you test #109?

**Comment by samme** [comment]
Closed by #109

**Comment by vpmedia** [comment]
Fix working, thanks!
